docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
+## 1.0 Introduction
+**Welcome to Quotesify!**
 
+Quotesify is a free desktop application to help you in your reading activities. With Quotesify, you can add
+books and the related quotes that you wish to remember. You can categorize your books and quotes by author,
+customized categories, and even rate your books. Quotesify also comes with a progress tracker just to improve
+your reading experience.
 
-## Product scope
-### Target user profile
+This guide will provide information on the design and implementation of Quotesify. It will help you get started
+on your journey of being a contributor to Quotesify. This guide will also explain the steps to test out the program,
+so that you will have a better understanding of the current status of Quotesify.
 
-{Describe the target user profile}
+## 2.0 Setting up
+1. Fork the Quotesify repo from [here](https://github.com/AY2021S1-CS2113T-T09-3/tp),
+and clone the fork to your computer.
+2. Open up your IDE (IntelliJ is highly recommended). If you are not at the welcome screen,
+click `File` > `Close Project` to close any existing project.
+3. Set up the correct JDK version for Gradle:
+    1. Click `Configure` > `Project Defaults` > `Project Structure`.
+    2. Click `New…` and find the directory where you saved your JDK.
+4. Click `Import Project`.
+5. Locate the *build.gradle* file and select it.
+6. Click `OK`.
+7. Click `Open as Project`.
+8. Click `OK` to accept all default settings.
+9. To verify the set up, locate the `Quotesify.java` file, right-click it and select `Run Quotesify.main()`.
+If the setup is correct, you should see something like this as shown below:
 
-### Value proposition
+```
+________                __                .__  _____       
+\_____  \  __ __  _____/  |_  ____   _____|__|/ ____\__.__.
+ /  / \  \|  |  \/  _ \   __\/ __ \ /  ___/  \   __<   |  |
+/   \_/.  \  |  (  <_> )  | \  ___/ \___ \|  ||  |  \___  |
+\_____\ \_/____/ \____/|__|  \___  >____  >__||__|  / ____|
+       \__>                      \/     \/          \/    
+Welcome to Quotesify v2.0!
 
-{Describe the value proposition: what problem does it solve?}
+What would you like to do with Quotesify?
+```
 
-## User Stories
+## 3.0 Design
+*`Note: All UML diagrams in this guide are stored in the images/ directory.`*
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+### 3.1 Architecture
+![Sequence Diagram for Architecture](images/SeqDiagram_Architecture.png)
+
+The architecture diagram displayed above describes the high-level design of Quotesify. Below details a brief
+description of each component shown.
+
+#### Main class: `Quotesify.java`
```

Comment:
Perhaps linking this to the corresponding GitHub file? [`Quotesify.java`](https://github.com/AY2021S1-CS2113T-T09-3/tp/blob/master/src/main/java/seedu/quotesify/Quotesify.java)